File: data/EXP01/python/2326/2326.output_2.py
# @lc app=leetcode id=2326 lang=python3
from typing import List, Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def spiralMatrix(self, m: int, n: int, head: Optional[ListNode]) -> List[List[int]]:
        matrix = [[-1] * n for _ in range(m)]
        
        values = []
        node = head
        while node:
            values.append(node.val)
            node = node.next
        
        left = 0
        right = n - 1
        top = 0
        bottom = m - 1
        
        row = 0
        col = 0
        dr = 0
        dc = 1

        for value in values:
            matrix[row][col] = value
            
            if row + dr < top or row + dr > bottom or col + dc < left or col + dc > right:
                if dr == 0 and dc == 1:
                    dr = 1
                    dc = 0
                elif dr == 1 and dc == 0:
                    dr = 0
                    dc = -1
                elif dr == 0 and dc == -1:
                    dr = -1
                    dc = 0
                elif dr == -1 and dc == 0:
                    dr = 0
                    dc = 1
            
            row += dr
            col += dc
        
        return matrix

File: data/EXP01/go/2326/2326.output_2.go
type ListNode struct {
    Val  int
    Next *ListNode
}

func spiralMatrix(m int, n int, head *ListNode) [][]int {
    matrix := make([][]int, m)
    for i := range matrix {
        matrix[i] = make([]int, n)
    }
    
    values := make([]int, 0)
    node := head
    for node != nil {
        values = append(values, node.Val)
        node = node.Next
    }
    
    left, right := 0, n-1
    top, bottom := 0, m-1
    index := 0
    
    for left <= right && top <= bottom {
        for i := left; i <= right && index < len(values); i++ {
            matrix[top][i] = values[index]
            index++
        }
        top++
        
        for i := top; i <= bottom && index < len(values); i++ {
            matrix[i][right] = values[index]
            index++
        }
        right--
        
        for i := right; i >= left && index < len(values); i-- {
            matrix[bottom][i] = values[index]
            index++
        }
        bottom--
        
        for i := bottom; i >= top && index < len(values); i-- {
            matrix[i][left] = values[index]
            index++
        }
        left++
    }
    
    for ; index < len(values); index++ {
        matrix[top][left] = values[index]
        if left < right {
            left++
        } else if top < bottom {
            top++
        }
    }
    
    for i := range matrix {
        for j := range matrix[i] {
            if matrix[i][j] == 0 {
                matrix[i][j] = -1
            }
        }
    }
    
    return matrix
}
